org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/tests/test.js

```diff
@@ -1,65 +1,108 @@
-/*jshint esversion: 6 */
-var opent2t = require('opent2t');
-var ava = require('ava');
-var config = require('./testconfig');
-var delay = require('delay')
-
-// This boilerplate code is copied from transpiled typescript AVA test code, which enables usage of 'yield' keyword. 
-var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
-    return new (P || (P = Promise))(function (resolve, reject) {
-        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
-        function rejected(value) { try { step(generator.throw(value)); } catch (e) { reject(e); } }
-        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
-        step((generator = generator.apply(thisArg, _arguments)).next());
-    });
-};
-
-function logAndValidate(p, name, expected) {
-    return p.then(result => {
-        if((expected !== undefined) && (result != expected))
-        {
-            throw name+"- Expected: "+ expected + ", Actual: "+ result;
-        }
-        else
-        {
-            console.log(name+ " (As expcted) : " + result);
-        }
-        return result;
-    }).catch(error => {
-        console.log("Test Error: " + error.message);
-        throw error;
-    });
-};
-
-console.log("Device Under Test -  Name: " + config.Device.name + "  Props: " + JSON.stringify(config.Device.props));
+var test = require('ava');
+var q = require('q');
+var OpenT2T = require('opent2t').OpenT2T;
+var config = require('./testConfig');
+
+console.log('Device Under Test -  Name: ' + config.Device.name + '  Props: ' + JSON.stringify(config.Device.props));
+var translatorPath = require('path').join(__dirname, '..');
 
 ///
 /// Run a series of tests to validate the translator
 ///
 
 // Get AmbientTemperature
-ava.test.serial("AmbientTemperature", (t) => __awaiter(this, void 0, void 0, function* () {
-    let device = yield opent2t.ThingAccessor.createTranslatorAsync(require("path").join(__dirname, ".."), "thingTranslator", config.Device);
-    t.is(typeof device, "object") && t.truthy(device);
-    let ambientTemperature = yield opent2t.ThingAccessor.getPropertyAsync(device, "org.opent2t.sample.thermostat.superpopular", "ambientTemperature");
-    console.log("*** ambientTemperature: " + ambientTemperature);
-    t.truthy(ambientTemperature);
-}));
+test.serial('AmbientTemperature', t => {
+    var deferred = q.defer();
+

```

Comment:
This and other tests can simplified with better use of promise chaining: Delete this line, and other references to `deferred` below, and the entire .catch() clause. Then put `return` at the beginning of lines 17 and 21. This method will then return a promise that is either resolved or rejected; if rejected then the stack trace will be printed by the test runner.